### PR TITLE
chore: update wascap & provider-archive deps

### LINKED ISSUE #596: Bump wascap, provider archive versions
Prior to working on the json schema issue we need to upgrade some dependencies. I tried doing this manually last week and couldn't get the dependency tree to sort out properly.

This shouldn't take long for someone more familiar with the wash deps tree

### PR DESCRIPTION
**Pull request by vados-cosmonic**
### TL;DR

You should find that all together:

- `wasmbus-rpc` must have it's `nkeys` version updated
- `wasmcloud-test-util` must have it's `wasmbus-rpc` version updated (w/ a `patch` for transitively used `nkeys` via `async-nats`)
- `wash` must have both `wasmbus-rpc` and `wasmcloud-test-util` updated to the locally built versions
- `wash`can then update `wascap` and `provider-archive`

### What this PR does

Update [`wascap`](https://crates.io/crates/wascap) and [`provider-archive`](https://crates.io/crates/provider-archive) deps which requires using an updated `nkeys` dependency (`v0.3.0`).

Unfortunately, due to `async-nats` using `v0.2.0` (even on the [latest version as of 2023/06/06](https://crates.io/crates/async-nats/0.29.0/dependencies)), version `0.2.0` is being pulled from various dependencies instead of `0.3.0`.

### Replicating this PR locally

To replicate this fix completely:

- clone `weld` (to influence `wasmbus-rpc` which is the `rpc-rs` folder)
- clone `wasmcloud-test` (to influence `wasmcloud-test-util`)
- update `nkeys` versions there (you will need a `[patch]` for `wasmcloud-test-util`)
  - You can copy the existing PRs noted above
- use your local versions as deps via the path syntax (`dep = { path = "..." }`)

If you'd like to see the tree for a given crate, use [`cargo tree`](https://doc.rust-lang.org/cargo/commands/cargo-tree.html).

## Related Issues

Resolves #596 

## Release Information

`next`

## Testing

Built on platform(s)
- [x] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

Tested on platform(s)
- [x] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

### CODE REVIEW

**Comment by vados-cosmonic** [comment]
At this point it looks like updating `async-nats` and staying on `nkeys` `0.2.0` for now might be the best way forward, so closing this
